_posts/2025-09-15-quarkus-a2a-multi-agent-content-creation.adoc

```diff
@@ -0,0 +1,244 @@
+
+==== Host
+
+In a terminal:
+
+[source,shell]
+----
+cd samples/python/hosts/content_creation
+uv run .
+----
+
+[NOTE]
+====
+As mentioned in the agent `README.md` files, don't forget to create a `.env` file for each agent with your
+`GOOGLE_API_KEY`. This is needed since the agents in this sample make use of Gemini. You can create a
+Google AI Studio API Key for free https://aistudio.google.com/[here].
+====
+
+=== Access the Content Creation Application
+
+Now that all of our agents are up and running, from your browser, navigate to http://localhost:8083.
+
+Try asking questions like:
+
+* Create a short, concise LinkedIn post about getting started with the Agent2Agent protocol
+* Create a short, upbeat X post about Quarkus LangChain4j
+
+image::UI.png[scaledwidth=100%]
+
+=== Swap Out an Agent
+
+One of the most powerful features of the A2A protocol is its interoperability. Let's see this in
+action by swapping out the TypeScript-based Content Editor agent for an equivalent agent written
+in Java.
+
+image::ContentCreationSwapped.png[scaledwidth=100%]
+
+First, stop the Host agent and the Content Editor agent.
+
+==== Start the Java Content Editor Agent
+
+In a terminal:
+
+[source,java]
+----
+cd samples/java/agents/content_editor
+----
+
+Follow the instructions in the `content_editor` https://github.com/a2aproject/a2a-samples/blob/main/samples/java/agents/content_editor/README.md[README.md] to start the Content Editor agent.
+
+==== Start the Host Agent
+
+In a terminal:
+
+[source,shell]
+----
+cd samples/python/hosts/content_creation
+uv run .
+----
+
+==== Access the Content Creation Application
+
+From your browser, navigate to http://localhost:8083 and try asking some questions.
+
+This time, the Host agent will seamlessly use the Java-based Content Editor agent for editing
+content instead of the TypeScript-based Content Editor agent. This flexibility is made possible
+because the A2A protocol is language-agnostic. This can be really useful for prototyping agents
+in one language to get things up and running quickly and then migrating to another language for
+a production environment.
+
+== Conclusion
+
+In this post, we saw how to use the A2A protocol to enable agents written in different programming
+languages and with different LLM frameworks to collaborate seamlessly to accomplish a goal. We also
+saw how easy it is to swap out one of the agents for an equivalent agent written in a different language.
+
+=== Further Reading
+
+* https://github.com/a2aproject/a2a-samples/blob/main/samples/python/hosts/content_creation/README.md[Content Creation Sample]
+* https://quarkus.io/blog/quarkus-a2a-java-0-3-0-alpha-release/[Getting Started with Quarkus and A2A Java SDK 0.3.0]
+* https://github.com/a2aproject/a2a-samples/tree/main/samples/java/agents[A2A Java SDK Samples]
+* https://github.com/a2aproject/a2a-java/blob/main/README.md[A2A Java SDK Documentation]
+* https://a2a-protocol.org/latest/specification/[A2A Specification]
+
```

Comment:
I'm wondering what the scalability model is here. Can't we have multiple instances of agents behind the same "card"? However, how would load balancing/session affinity work?